In [37]:
import numpy as np
import scipy.stats as stats
data = [-23, -16, -3, -9, -1, 13, -16, 15, 31, -22, -14, -25]
#use t test since n < 30

mu = 0
size = len(data)
mean = np.mean(data)
sdt = np.std(data, ddof = 1)
se = sdt / (np.sqrt(size))
t = mean / se
tc = stats.t.ppf(1 - ((1 - .9) / 2), df = size - 1) #90% use t test
maxTc = mean + (tc * se)
minTc = mean - (tc * se)

print("Mean:", mean)
print("Standard Error:", se)
print("T-test statistic:", t)
print("Interval: (", minTc, "," , maxTc, ")")

Mean: -5.833333333333333
Standard Error: 5.062987100895912
T-test statistic: -1.152152517295789
Interval: ( -14.925875005124801 , 3.2592083384581363 )


In [38]:

tc = stats.t.ppf(1 - ((1 - .95) / 2), df = size - 1) #95% use t test
maxTc = mean + (tc * se)
minTc = mean - (tc * se)
print("Mean:", mean)
print("Standard Error:", se)
print("T-test statistic:", t)
print("Interval: (", minTc, "," , maxTc, ")")

Mean: -5.833333333333333
Standard Error: 5.062987100895912
T-test statistic: -1.152152517295789
Interval: ( -16.976892808096625 , 5.31022614142996 )


In [39]:
sdt = 16.836
se = sdt / (np.sqrt(size))
z = mean / se
zc = stats.norm.ppf(1 - ((1 - .95) / 2)) #95% use normal distribution cause using actual population standard deviation
maxTc = mean + (zc * se)
minTc = mean - (zc * se)
print("Mean:", mean)
print("Standard Error:", se)
print("Z-test statistic:", z)
print("Interval: (", minTc, "," , maxTc, ")")

Mean: -5.833333333333333
Standard Error: 4.860134566038269
Z-test statistic: -1.200241115564083
Interval: ( -15.359022042786545 , 3.69235537611988 )


In [40]:
#need to find percent where maxTc is 0 (team is always losing) and that value is the confidence that the team will expect
#to lose on average

confidence = 1 #100% confidence
while(maxTc > 0):
    confidence -= 0.00001
    tc = stats.norm.ppf(1 - ((1 - confidence) / 2)) #95% use normal distribution cause using actual population standard deviation
    maxTc = mean + (tc * se)
    minTc = mean - (tc * se)
    
print("Interval: (", minTc, "," , maxTc, ")")
print("Confidence Value:", confidence)

Interval: ( -11.66661298382914 , -5.36828375272691e-05 )
Confidence Value: 0.769950000001047


Writeup

2.1
For 90% confidence interval:
Using t-test because n < 30.
Mean: -5.833333333333333
Standard Error: 5.062987100895912
T-test statistic: -1.152152517295789
Interval: (-14.925875005124801 , 3.2592083384581363)

2.2
For 95% confidence interval:
Using t-test because n < 30.
Mean: -5.833333333333333
Standard Error: 5.062987100895912
T-test statistic: -1.152152517295789
Interval: (-16.976892808096625 , 5.31022614142996)

The main differences between 95% confidence interval and 90% confidence interval is that the 95% confidence interval has a wider range of values. This is expected because to have a higher confidence in a result, your range of values will be higher to have a higher chance of being correct.

2.3
For 95% confidence interval
Using z-test because it is a normal distribution as we know the standard deviation.
Mean: -5.833333333333333
Standard Error: 4.860134566038269
Z-test statistic: -1.200241115564083
Interval: (-15.359022042786545 , 3.69235537611988)

The main difference between using the z-test and t-test is that when using the z-test we have a lower standard error along with a lower upper limit for our confidence interval. 

2.4
Using a while loop to increment the confidence percentage value we can find a confidence value where the upper limit for the confidence interval is less than 0, which is the equivalent to the team losing.
Interval: ( -11.66661298382914 , -5.36828375272691e-05 )
Confidence Value: 0.769950000001047
We have a confidence value of about 77% that the team will lose on average. 